sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreChangelog.scala

```diff
@@ -388,6 +451,38 @@ class StateStoreChangelogReaderV2(
   }
 }
 
+class StateStoreChangelogReaderV3(
+     fm: CheckpointFileManager,
+     fileToRead: Path,
+     compressionCodec: CompressionCodec)
+  extends StateStoreChangelogReaderV1(fm, fileToRead, compressionCodec) {
+
+  override def version: Short = 3
+
+  verifyVersion()
```

Comment:
don't call these individually in all the formats... Error prone for future versions

Reply:
Yes I agree.. but for the same reason above (only v1 does not have version written in the first line), we need to handle this in a case by case fashion